Public/Src/Tools/Tool.MsBuildGraphBuilder/PropertyTrackingLogger.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BuildXL.Utilities.Collections;
+using Microsoft.Build.Framework;
+
+namespace ProjectGraphBuilder
+{
+    internal class PropertyTrackingLogger : ILogger
+    {
+        private readonly HashSet<string> _variablesRead = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private bool _nonTrackingSdkResolversExist = false;
+
+        public IEnumerable<string> PotentialEnvironmentVariablesReads
+        {
+            get
+            {
+                // Return a copy of what's there to minimize threading issues.
+                lock (_variablesRead)
+                {
+                    return _variablesRead.ToArray();
+                }
+            }
+        }
+
+        public void Initialize(IEventSource eventSource)
+        {
+            eventSource.AnyEventRaised += (sender, args) =>
+                                          {
+                                              if (args is EnvironmentVariableReadEventArgs evrArgs)
+                                              {
+                                                  lock(_variablesRead)
+                                                  {
+                                                      _variablesRead.Add(evrArgs.EnvironmentVariableName);
+                                                  }
+                                              }
+                                              else if (args is UninitializedPropertyReadEventArgs upArgs)
+                                              {
+                                                  lock(_variablesRead)
+                                                  {
+                                                      _variablesRead.Add(upArgs.PropertyName);
+                                                  }
+                                              }
+                                              else if (args is BuildWarningEventArgs warnArgs && warnArgs.Code == "MSB4263")
+                                              {
+                                                  // We just want to know whether or not we can trust the above tracking data.
+                                                  _nonTrackingSdkResolversExist = true;
+                                              }

```

Comment:
So when tracking is not available, can we get a reason for why? So we can print that as a warning on bxl side

Reply:
Added a call to reporter.ReportMessage when there is a failure. Seen in latest commit.